store/postgres/src/subgraph_store.rs

```diff
@@ -1147,23 +1147,67 @@ impl SubgraphStoreInner {
     }
 }
 
+const STATE_ENS_NOT_CHECKED: u8 = 0;
+const STATE_ENS_EMPTY: u8 = 1;
+const STATE_ENS_NOT_EMPTY: u8 = 2;
+
+/// EnsLookup reads from a rainbow table store in postgres that needs to be manually
+/// loaded. To avoid unnecessary database roundtrips, the empty table check is lazy
+/// and will not be retried. Once the table is checked, any subsequent calls will
+/// just used the stored result.
 struct EnsLookup {
     primary: ConnectionPool,
+    // In order to keep the struct lock free, we'll use u8 for the status:
+    // 0 - Not Checked
+    // 1 - Checked - empty
+    // 2 - Checked - non empty
+    state: AtomicU8,
```

Comment:
This caching has the downside of requiring a graph node reset after the names are loaded.

I don't think this is worth it as an optimization, the `fn is_ens_data_empty` query is cheap and it is run only if the name lookup returned empty.